docs/fsharp-core-notes.md

```diff
@@ -1,126 +1,118 @@
+## Guidance for everyone else
 
-### Libraries should target lower versions of FSharp.Core
+If you're not authoring packages for distribution, you have a lot less to worry about.
 
-F# ecosystem libraries should generally target the *earliest, most portable* profile of FSharp.Core feasible, within reason.
+If you are distributing library code across a private organization as if it were a NuGet package, please see the above guidance, as it likely still applies. Otherwise, the below guidance applies.
 
-If your library is part of an ecosystem, it can be helpful to target the _earliest, most widespread language version_ 
-and the _earliest_ and _most supported_ profiles of the .NET Framework feasible.
+### Application authors don't have to explicitly reference FSharp.Core
 
-The version you choose should be based on the minimum F# language version you want to support. The minimum FSharp.Core version for each language version is listed below:
+In general, applications can always just use the latest FSharp.Core bundled in the SDK they are built with.
 
-|Minimum F# language version|Minimum FSharp.Core version|
-|------------------------------|------------------------------|
-|F# 4.1|4.3.4|
-|F# 4.5|4.5.2|
-|F# 4.6|4.6.2|
-|F# 4.7|4.7.2|
-|F# 5.0|5.0.0|
+### C# projects referencing F# projects may need to pin FSharp.Core
+
+You can reference an F# project just fine without needing to be explicit about an FSharp.Core reference when using non-legacy projects. References flow transitively, so even if you need to use types directly from FSharp.Core (which you probably shouldn't do anyways) it will pick up the right types from the right assembly.
 
-A good choice for libraries is to target `netstandard2.0` and FSharp.Core 4.7.2.
+If you do have an explicit FSharp.Core reference in your C# project that you **need**, you should pin your FSharp.Core reference across your entire codebase. Being in a mixed pinned/non-pinned world is difficult to keep straight over a long period of time.
 
-    <PackageReference Update="FSharp.Core" Version="4.7.2" />
+## Guidance for legacy projects, compilers, and tools
 
-For "libraries" that are effectively part of an application, you can just target
-the latest language version and the framework you're using in your application.
+Lastly, if you are using legacy F# projects (non-SDK) and/or have an older toolset, the following guidance applies.
 
-### Applications should target higher versions of FSharp.Core
+### Consider upgrading
 
-F# applications should generally use the *highest* language version and the most *platform-specific* version of FSharp.Core.
+Yes, really. The old project system that manages legacy projects is not that good, the compiler is older and unoptimized, tooling is not as responsive, etc. You will really have a much better life if you upgrade. Try out the `try-convert` tool to do that: https://github.com/dotnet/try-convert
```

Comment:
What does this mean?  "compiler is older and unoptimized"

Reply:
It's significantly unoptimized compared to today. If the same compiler were used to support a large solution it would likely make VS go OOM. Compile-times are also severely degraded. Just between .NET Core 2.1 and .NET 5, the compiler can compile FSharpPlus 2x as fast: https://devblogs.microsoft.com/dotnet/announcing-f-5/#improved-compiler-performance

Reply:
By upgrading are we suggesting, upgrading the projects to use net Sdk projects, or upgrading the version of vs the developer is using to the latest vs?

I think the tryconvert tool, is for upgrading the project file to dotnet sdk.  The compiler version and tools would be unimpacted by the use of try convert.  The desktop and coreclr versions of the FSharp compiler deployed with Visual Studio, are built from the same codebase, modulo some ifdefs, and are broadly speaking equally performant.

Upgrading the project to the dotnet sdk, is very advantageous, the project files are simpler with better defaults and support PackageReferences, the net sdk project system also scales better to larger solutions.